### ISSUE TITLE
Memory-Leak in WriteOnePNGImage() coders/png.c

### ISSUE THREAD

**Issue by lcatro** [ImageMagick/ImageMagick#569]
Memory Leak File Link : https://raw.githubusercontent.com/lcatro/My_PoC/master/ImageMagick/memory-leak_output_png_WriteOnePNGImage

Trigger Command : ./magick convert memory-leak_output_png_WriteOnePNGImage output.png

Leak Detail : 
```
fuzzing@ubuntu:~/fuzzing/ImageMagick/utilities$ ./magick convert memory-leak_output_png_WriteOnePNGImage output.png
convert: UnableToOpenConfigureFile `magic.xml' @ warning/configure.c/GetConfigureOptions/715.

=================================================================
==62838==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7fa46b846602 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x98602)
    #1 0x7fa46ad4f981 in AcquireMagickMemory MagickCore/memory.c:464
    #2 0x7fa46ae6decf in AcquireStringInfo MagickCore/string.c:174
    #3 0x7fa46ae6e979 in CloneStringInfo MagickCore/string.c:330
    #4 0x7fa46b176e9f in WriteOnePNGImage coders/png.c:11581
    #5 0x7fa46b17bd2c in WritePNGImage coders/png.c:12536
    #6 0x7fa46abab1c6 in WriteImage MagickCore/constitute.c:1114
    #7 0x7fa46ababe42 in WriteImages MagickCore/constitute.c:1333
    #8 0x7fa46a3fe3eb in ConvertImageCommand MagickWand/convert.c:3280
    #9 0x7fa46a4f6d98 in MagickCommandGenesis MagickWand/mogrify.c:183
    #10 0x4017f1 in MagickMain utilities/magick.c:149
    #11 0x4019d2 in main utilities/magick.c:180
    #12 0x7fa469c6b82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)

Indirect leak of 4116 byte(s) in 1 object(s) allocated from:
    #0 0x7fa46b846602 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x98602)
    #1 0x7fa46ad4f981 in AcquireMagickMemory MagickCore/memory.c:464
    #2 0x7fa46ad4f9d5 in AcquireQuantumMemory MagickCore/memory.c:537
    #3 0x7fa46ae6e084 in AcquireStringInfo MagickCore/string.c:182
    #4 0x7fa46ae6e979 in CloneStringInfo MagickCore/string.c:330
    #5 0x7fa46b176e9f in WriteOnePNGImage coders/png.c:11581
    #6 0x7fa46b17bd2c in WritePNGImage coders/png.c:12536
    #7 0x7fa46abab1c6 in WriteImage MagickCore/constitute.c:1114
    #8 0x7fa46ababe42 in WriteImages MagickCore/constitute.c:1333
    #9 0x7fa46a3fe3eb in ConvertImageCommand MagickWand/convert.c:3280
    #10 0x7fa46a4f6d98 in MagickCommandGenesis MagickWand/mogrify.c:183
    #11 0x4017f1 in MagickMain utilities/magick.c:149
    #12 0x4019d2 in main utilities/magick.c:180
    #13 0x7fa469c6b82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)

SUMMARY: AddressSanitizer: 4148 byte(s) leaked in 2 allocation(s).

```

**Comment by mikayla-grace** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ http://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by bastien-roucaries** [comment]
Is is a V6 problem ?

**Comment by glennrp** [comment]
IM6 was fixed the same day, commit 0a80c9e5f293a8de51011ac784ac52b96932c08f

**Comment by parr0tr1ver** [comment]
It should fixed in V7 by commit e436a58
